### ISSUE TITLE
Get stage

### ISSUE THREAD

**Issue by hesamfarhang** [konvajs/react-konva#255]
How get stage without event trigger?

`render(){ (
    <Stage>
        <Layer></Layer>
    </Stage>
) }`
for example in componentDidUpdate
`componentDidMount(){
    let stage= ?
}`

**Comment by lavrton** [comment]
```javascript
class App extends React.Component {
  componentDidMount() {
    // log stage react wrapper
    console.log(this.stageRef);
    // log Konva.Stage instance
    console.log(this.stageRef.getStage());
  }
  render() {
    return <Stage ref={ref => { this.stageRef = ref; }} width={300} height={300} />;
  }
}
```

**Comment by damiangreen** [comment]
where are you declaring the ref here , and how, useRef/ createRef?
I don't seem to be able to get this to work

**Comment by lavrton** [comment]
```js
const App = () => {
  const stageRef = React.useRef();

  React.useEffect(() => {
    // logs `Konva.Stage` node instance
    console.log(stageRef.current);
  })

  return <Stage ref={stageRef} />;
}
```

**Comment by gabtorr** [comment]
Hi, first i want to thank you for you great work and your great help for community. I just having an issue recently. I use the following code provided by @lavrton to export my stage, the issue came when i drag the elements on canvas the export image always shows the initial position of the uploaded image rendered, as well of the other elements dragged in the canvas. I need to add something else for capture the picture of the last position of elements?

> ```js
> const App = () => {
>   const stageRef = React.useRef();
> 
>   React.useEffect(() => {
>     // logs `Konva.Stage` node instance
>     console.log(stageRef.current);
>   })
> 
>   return <Stage ref={stageRef} />;
> }
> ```

**Comment by lavrton** [comment]
> I need to add something else for capture the picture of the last position of elements?

No, if you export stage it will capture the last position that you see on the screen. Can you make a demo of your issue?

**Comment by gabtorr** [comment]
> > I need to add something else for capture the picture of the last position of elements?
> 
> No, if you export stage it will capture the last position that you see on the screen. Can you make a demo of your issue?

Thanks for your quick response! ... Sure, here si a [codesandbox](https://codesandbox.io/s/fragrant-breeze-0yi3c?file=/src/App.js) , the component that i had its like the same  that is on the sandbox. Also use this [link](https://0yi3c.csb.app/) to see the export on console.

**Comment by lavrton** [comment]
That looks like a bug. I will take a look.

**Comment by gabtorr** [comment]
> That looks like a bug. I will take a look.

Hi, i find a solution, for my case. On the custom URLImage component i set the Konva.Image component to draggable, instead of the layer. Due to that the rendered image is draggable and when exporting shows the image in the position that its droped. Probably with the layer draggable should work but thats the solution i find. Anyways thank for your help and taking the time to readme.

**Comment by lavrton** [comment]
Anyway, I fixed it with a new `konva@7.1.6`. Thanks for the report.
